ts/Extensions/MouseWheelZoom/MouseWheelZoom.ts

```diff
@@ -68,196 +72,211 @@ const optionsToObject = (
 /**
  * @private
  */
-const fitToBox = function (
-    inner: BBoxObject,
-    outer: BBoxObject
-): BBoxObject {
-    if (inner.x + inner.width > outer.x + outer.width) {
-        if (inner.width > outer.width) {
-            inner.width = outer.width;
-            inner.x = outer.x;
-        } else {
-            inner.x = outer.x + outer.width - inner.width;
-        }
-    }
-    if (inner.width > outer.width) {
-        inner.width = outer.width;
-    }
-    if (inner.x < outer.x) {
-        inner.x = outer.x;
-    }
-
-    // y and height
-    if (inner.y + inner.height > outer.y + outer.height) {
-        if (inner.height > outer.height) {
-            inner.height = outer.height;
-            inner.y = outer.y;
+const fitToRange = (
+    outerStart: number,
+    outerWidth: number,
+    innerStart: number,
+    innerWidth: number
+): {
+    rangeStart: number,
+    rangeWidth: number
+} => {
+    if (innerStart + innerWidth > outerStart + outerWidth) {
+        if (innerWidth > outerWidth) {
+            innerWidth = outerWidth;
+            innerStart = outerStart;
         } else {
-            inner.y = outer.y + outer.height - inner.height;
+            innerStart = outerStart + outerWidth - innerWidth;
         }
     }
-    if (inner.height > outer.height) {
-        inner.height = outer.height;
+    if (innerWidth > outerWidth) {
+        innerWidth = outerWidth;
     }
-    if (inner.y < outer.y) {
-        inner.y = outer.y;
+    if (innerStart < outerStart) {
+        innerStart = outerStart;
     }
 
-    return inner;
+    return {
+        rangeStart: innerStart,
+        rangeWidth: innerWidth
+    };
 };
 
 let wheelTimer: number,
     startOnTick: boolean|undefined,
     endOnTick: boolean|undefined;
 
 /**
- * @private
- */
-const zoomBy = function (
+* @private
+*/
+
+const zoomOnDirection = function (
     chart: Chart,
+    axis: Axis,
+    mousePos: number,
     howMuch: number,
-    centerXArg: number,
-    centerYArg: number,
-    mouseX: number,
-    mouseY: number,
-    options: MouseWheelZoomOptions
-): boolean {
-    const xAxis = chart.xAxis[0],
-        yAxis = chart.yAxis[0],
-        yOptions = yAxis.options,
-        type = pick(options.type, chart.zooming.type, ''),
-        zoomX = /x/.test(type),
-        zoomY = /y/.test(type);
+    centerArg: number
+) : boolean {
+    const isXAxis = axis.isXAxis,
+        axisOptions = axis.options;
+
+    let hasZoomed = false,
+        fixToDir;
 
-    let hasZoomed = false;
+    if (defined(axis.max) && defined(axis.min) &&
+        defined(axis.dataMax) && defined(axis.dataMin)) {
 
-    if (defined(xAxis.max) && defined(xAxis.min) &&
-        defined(yAxis.max) && defined(yAxis.min) &&
-        defined(xAxis.dataMax) && defined(xAxis.dataMin) &&
-        defined(yAxis.dataMax) && defined(yAxis.dataMin)) {
+        if (isXAxis) {
+            fixToDir = mousePos ? ((mousePos - axis.pos) / axis.len) : 0.5;
```

Comment:
I'd go for splitting xAxis- and yAxis-specific functionalities out of this function. The fixToDir value gets calculated a bit differently, so I'd leave it out of the scope of this function. It will be also a bit easier to read.